core/src/main/java/org/infinispan/container/entries/DeltaAwareCacheEntry.java

```diff
@@ -168,9 +180,7 @@ public boolean isNull() {
    public final void commit(final DataContainer<K, DeltaAware> container, final Metadata metadata) {
       //If possible, we now ensure copy-on-write semantics. This way, it can ensure the correct transaction isolation.
       //note: we want to merge/copy to/from the data container value.
-      container.compute(key, new DataContainer.ComputeAction<K, DeltaAware>() {
-         @Override
-         public InternalCacheEntry<K, DeltaAware> compute(K key, InternalCacheEntry<K, DeltaAware> oldEntry, InternalEntryFactory factory) {
+      PersistenceUtil.loadAndComputeInDataContainer(container, persistenceManager, key, ctx, timeService, (key, oldEntry, factory) -> {
```

Comment:
I was thinking that in non-tx mode too, the wrapping happens after the locking, so you can't have other writes modifying the entry. Of course, I had forgotten this is for `applyDelta()`, which works with partial locks.